modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Generate.java

```diff
@@ -240,6 +240,9 @@
     @Option(name = {"--generate-alias-as-model"}, title = "generate alias (array, map) as model", description = CodegenConstants.GENERATE_ALIAS_AS_MODEL_DESC)
     private Boolean generateAliasAsModel;
 
+    @Option(name = {"--discriminator-explicit-mapping-verbose"}, title = "explicitly define a discriminator on all models which have one, add verbose mapping to that discriminator, validate discriminators", description = CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE_DESC)
```

Comment:
This option needs clarification since it's currently doing a lot.

1. Can we remove the "verbose" since that's not really relevant to the user?
2. The term `mapping` would indicate to users who are familiar with our options that this would take a string in the format `key=value`, which is not the case.
3. The option is triggering OpenAPI 3.x logic, but doesn't indicate that clearly here.

Might I recommend changing this to something more like:

```
@Option(name = {"--legacy-discriminators"}, title = "Support legacy logic for evaluating discriminators", description = CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE_DESC)
private Boolean legacyDiscriminators;
```

This value would need to default to `true` until the new logic becomes default (I don't recall if it's now default or not).

The value would then have to be applied to CodegenConfigurator, which is not done in `Generate.java` here. The same default would have to be done in the Maven and Gradle plugins, with a task to add the feature option to those. We also have a new SBT plugin (pending first release) where this would have to be added.

Reply:
This was mistakenly commented in the wrong PR. I am marking it resolved.